br/pkg/lightning/backend/external/writer.go

```diff
@@ -509,9 +472,61 @@ func (w *Writer) flushKVs(ctx context.Context, fromClose bool) (err error) {
 	w.kvBuffer.Reset()
 	w.rc.reset()
 	w.batchSize = 0
+	w.currentSeq++
 	return nil
 }
 
+func (w *Writer) flushSortedKVs(ctx context.Context) (string, string, error) {
+	logger := logutil.Logger(ctx).With(
+		zap.String("writer-id", w.writerID),
+		zap.Int("sequence-number", w.currentSeq),
+	)
+	writeStartTime := time.Now()
+	dataFile, statFile, dataWriter, statWriter, err := w.createStorageWriter(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	kvStore, err := NewKeyValueStore(ctx, dataWriter, w.rc)
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, pair := range w.kvLocations {
+		err = kvStore.addEncodedData(w.kvBuffer.GetSlice(pair))
+		if err != nil {
+			return "", "", err
+		}
+	}
+
+	kvStore.Close()
+	encodedStat := w.rc.encode()
+	statSize := len(encodedStat)
+	_, err = statWriter.Write(ctx, encodedStat)
+	if err != nil {
+		return "", "", err
+	}
+	err = dataWriter.Close(ctx)
+	if err != nil {
+		return "", "", err
+	}
+	err = statWriter.Close(ctx)
+	if err != nil {
+		return "", "", err
+	}
```

Comment:
close them even on previous err path?

Reply:
OK, `close` means creating the file in object storage, but I think it's more reasonable for other backends